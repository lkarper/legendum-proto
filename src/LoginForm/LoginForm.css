@import url('https://fonts.googleapis.com/css2?family=Cinzel+Decorative:wght@400;700&family=Crimson+Text&display=swap');

.LoginForm__container {
    display: flex;
    flex-direction: column;
    margin: 10px;
    padding: 5px;
    font-family: 'Crimson Text', serif;
}

.LoginForm__h2 {
    font-weight: 400;
}

.LoginForm__input-label-container {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    padding: 5px;
}

.LoginForm__input-label-container input {
    width: 75%;
    font-family: 'Crimson Text', serif;
}

.LoginForm__error-container {
    height: 25px;
}

.LoginForm__error {
    font-family: 'Crimson Text', serif;
    color: red;
    font-weight: bold;
    font-size: 20px;
    margin: 10px;
}

.LoginForm__submit {
    width: 100px;
    height: 50px;
    font-size: 20px;
    margin: 10px;
}