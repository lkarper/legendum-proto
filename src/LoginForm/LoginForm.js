import React from 'react';

const LoginForm = (props) => {

    const { userName, password, setUserName, setPassword, handleLogin, error } = props;

    return (
        <div>
            <h2>Login</h2>
            <form onSubmit={(e) => handleLogin(e)}>
                <label htmlFor="user-name">Username:</label>
                <input
                    type="text"
                    id="user-name"
                    name="user-name"
                    value={userName}
                    onChange={(e) => setUserName(e.target.value)}
                    autoComplete="username"
                    required 
                />
                <label htmlFor="password">Password:</label>
                <input 
                    type="password"
                    id="password"
                    name="password"
                    value={password}
                    onChange={(e) => setPassword(e.target.value)}
                    autoComplete="current-password"
                    required
                />
                <div role='alert'>
                    {error && <p className='red'>{error}</p>}
                </div>
                <button
                    type="submit"
                    disabled={!userName || !password}
                >
                    Login
                </button>
            </form>            
        </div>
    );
}

export default LoginForm;