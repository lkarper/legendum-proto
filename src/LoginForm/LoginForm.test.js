import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import LoginForm from './LoginForm';

describe('LoginForm component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<LoginForm />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<LoginForm />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
