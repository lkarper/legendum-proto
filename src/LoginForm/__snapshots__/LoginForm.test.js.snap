// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component renders the UI as expected 1`] = `
<LoginForm
  error=""
  handleLogin={[Function]}
  password=""
  setPassword={[Function]}
  setUserName={[Function]}
  userName=""
>
  <div
    className="LoginForm__container"
  >
    <h2
      className="LoginForm__h2"
    >
      Login
    </h2>
    <form
      className="LoginForm__form"
      onSubmit={[Function]}
    >
      <div
        className="LoginForm__input-label-container"
      >
        <label
          htmlFor="user-name"
        >
          Username:
        </label>
        <input
          autoComplete="username"
          id="user-name"
          name="user-name"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="LoginForm__input-label-container"
      >
        <label
          htmlFor="password"
        >
          Password:
        </label>
        <input
          autoComplete="current-password"
          id="password"
          name="password"
          onChange={[Function]}
          required={true}
          type="password"
          value=""
        />
      </div>
      <div
        className="LoginForm__error-container"
        role="alert"
      />
      <button
        className="LoginForm__submit button"
        disabled={true}
        type="submit"
      >
        Login
      </button>
    </form>
  </div>
</LoginForm>
`;
